### PR TITLE
Granular CRN for image datasources and resources

### PR DESCRIPTION
**Pull request by Alexander-Kita**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

pi_catalog_images (d):

```
=== RUN   TestAccIBMPICatalogImagesDataSourceBasic
--- PASS: TestAccIBMPICatalogImagesDataSourceBasic (52.36s)
PASS
```

```
=== RUN   TestAccIBMPICatalogImagesDataSourceVTL
--- PASS: TestAccIBMPICatalogImagesDataSourceVTL (51.41s)
PASS
```

```
=== RUN   TestAccIBMPICatalogImagesDataSourceSAP
--- PASS: TestAccIBMPICatalogImagesDataSourceSAP (74.17s)
PASS
```

```
=== RUN   TestAccIBMPICatalogImagesDataSourceSAPAndVTL
--- PASS: TestAccIBMPICatalogImagesDataSourceSAPAndVTL (61.02s)
PASS
```

pi_image (d):

```
=== RUN   TestAccIBMPIImageDataSource_basic
--- PASS: TestAccIBMPIImageDataSource_basic (17.08s)
PASS
```

pi_images (d):

```
=== RUN   TestAccIBMPIImagesDataSource_basic
--- PASS: TestAccIBMPIImagesDataSource_basic (28.87s)
PASS
```

pi_image (r):

```
=== RUN   TestAccIBMPIImagebasic
--- PASS: TestAccIBMPIImagebasic (40.15s)
PASS
```

(Used private bucket)
```
=== RUN   TestAccIBMPIImageCOSPublicImport
--- PASS: TestAccIBMPIImageCOSPublicImport (1227.62s)
PASS
```

```
=== RUN   TestAccIBMPIImageBYOLImport
--- PASS: TestAccIBMPIImageBYOLImport (1817.78s)
PASS
```

### CODE REVIEW

File: ibm/service/power/data_source_ibm_pi_catalog_images.go [hunk 1]
```diff
@@ -59,6 +59,11 @@ func DataSourceIBMPICatalogImages() *schema.Resource {
 							Description: "Date of image creation",
 							Type:        schema.TypeString,
 						},
+						Attr_CRN: {
+							Computed:    true,
+							Description: "CRN of resource.",
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_catalog_images.go:64]:
```suggestion
							Description: "CRN of this resource.",
```

File: ibm/service/power/data_source_ibm_pi_catalog_images.go [hunk 2]
```diff
@@ -159,6 +164,9 @@ func dataSourceIBMPICatalogImagesRead(ctx context.Context, d *schema.ResourceDat
 		if i.CreationDate != nil {
 			image[Attr_CreationDate] = i.CreationDate.String()
 		}
+
+		image[Attr_CRN] = i.Crn
+
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_catalog_images.go:169]: Could you add a check for if we have that value or not? And then set it.

File: ibm/service/power/data_source_ibm_pi_image.go [hunk 3]
```diff
@@ -37,6 +37,11 @@ func DataSourceIBMPIImage() *schema.Resource {
 				Description: "The CPU architecture that the image is designed for. ",
 				Type:        schema.TypeString,
 			},
+			Attr_CRN: {
+				Computed:    true,
+				Description: "CRN of image.",
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_image.go:42]:
```suggestion
				Description: "CRN of the image.",
```

File: ibm/service/power/data_source_ibm_pi_images.go [hunk 4]
```diff
@@ -34,6 +34,11 @@ func DataSourceIBMPIImages() *schema.Resource {
 				Description: "List of all supported images.",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						Attr_CRN: {
+							Computed:    true,
+							Description: "CRN of image.",
```
Comment (michaelkad) [inline ibm/service/power/data_source_ibm_pi_images.go:39]:
```suggestion
							Description: "CRN of the image.",
```

File: ibm/service/power/resource_ibm_pi_image.go [hunk 5]
```diff
@@ -240,12 +252,17 @@ func resourceIBMPIImageCreate(ctx context.Context, d *schema.ResourceData, meta
 		bucketImageFileName := d.Get(helpers.PIImageBucketFileName).(string)
 		bucketRegion := d.Get(helpers.PIImageBucketRegion).(string)
 		bucketAccess := d.Get(helpers.PIImageBucketAccess).(string)
+		var userTags []string
+		if tags, ok := d.GetOk(Arg_UserTags); ok {
+			userTags = flex.ExpandStringList(tags.([]interface{}))
+		}
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_image.go:258]: Could add this `body.UserTags= userTags`? And check `if len() >0` first

**Review by michaelkad — CHANGES_REQUESTED** [review]
Few comments.

**Review by ismirlia — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
Opened in main repo

**Pull request closed without merging**